File: my-folder/0020-valid-parentheses/solution.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        open_brackets = ['(', '{', '[']
        close_brackets = [')', '}', ']']
        stack = []
        for char in s:
            if char in open_brackets:
                stack.append(char)
            else:
                if (len(stack) == 0):
                    return False
                prev_char = stack.pop()
                if open_brackets.index(prev_char) != close_brackets.index(char):
                    return False
        return len(stack) == 0

File: my-folder/0020-valid-parentheses/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {

    const stack = [];
    const openBrackets = ["(", "{", "["];
    const closeBrackets = [")", "}", "]"];

    for (const bracket of s) {
        if (openBrackets.includes(bracket)) {
            stack.push(bracket);
        } else {
            const openBracket = stack.pop();
            if (openBrackets.indexOf(openBracket) !== closeBrackets.indexOf(bracket)) {
                return false;
            }
        }
    }
    return stack.length === 0;
};
